### PR TITLE
Make WiredList Start/Stop race-free

### PR DESCRIPTION
**Pull request by vdarulis**
**What this PR does / why we need it**:

Fixes race detector failures in TestRoundtrip by making sure the update/doneCh are not mutated unsafely in different goroutines.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

**Comment by robskillington** [comment]
Hey thanks for this, I think though that we want to protect against enqueuing to a closed channel too, so maybe using an Rlock to protect against publishing to the channel is the best approach?
https://github.com/m3db/m3/pull/2606

There should be zero contention (since we don't close until end of process) so should not add much overhead.

**Comment by vdarulis** [comment]
> Hey thanks for this, I think though that we want to protect against enqueuing to a closed channel too, so maybe using an Rlock to protect against publishing to the channel is the best approach?

Sure thing, this should merge cleanly with #2606 and do the _right_ thing, since channel is nil'ed on the wlock.
I didn't want to change semantics in this PR.

**Comment by vdarulis** [comment]
Oops nvm it was a mutex before, not RWMutex

File: src/dbnode/storage/block/wired_list.go [hunk 1]
```diff
@@ -204,6 +224,11 @@ func (l *WiredList) Stop() error {
 //
 // We use a channel and a background processing goroutine to reduce blocking / lock contention.
 func (l *WiredList) BlockingUpdate(v DatabaseBlock) {
+	l.mu.RLock()
+	defer l.mu.RUnlock()
```
Comment (robskillington) [inline src/dbnode/storage/block/wired_list.go:228]: Need protection around NonBlockingUpdate too.

**Comment by vdarulis** [comment]
abandoning in favor of #2606

**Pull request closed without merging**